<launch>
  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find wpr_simulation)/worlds/ROS-Academy.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="recording" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <!-- Spawn a robot into Gazebo -->
  <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-file $(find wpr_simulation)/urdf/wpb_home.gazebo -urdf -model wpb_home" />

  <arg name="model" default="$(find wpb_home_bringup)/urdf/wpb_home_mani.urdf"/>
  <arg name="gui" default="true" />
  <arg name="rvizconfig" default="$(find wpb_home_tutorials)/rviz/nav.rviz" />

  <param name="robot_description" command="$(find xacro)/xacro.py $(arg model)" />
  <param name="use_gui" value="$(arg gui)"/> 

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

  <!--- Run servers -->
  <include file="$(find wpb_home_behaviors)/launch/all_servers.launch" />

  <!-- Run the map server -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find wpb_home_tutorials)/maps/map.yaml"/>

  <!--- Run AMCL -->
  <include file="$(find wpb_home_tutorials)/nav_lidar/amcl_omni.launch" />

  <!-- Map tools  -->
  <node pkg="waterplus_map_tools" type="wp_manager" name="wp_manager" output="screen"/>
    
  <!-- Run lidar filter -->
  <node pkg="wpb_home_bringup" type="wpb_home_lidar_filter" name="wpb_home_lidar_filter">
    <param name="pub_topic" value="/scan"/>
  </node>

  <node pkg="xfyun_waterplus" type="iat_node" name="xfyun_iat_node" output="screen">
        <param name="cn" type="bool" value="true"/> 
  </node>

  <!--- Run move base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base"  output="screen">
    <rosparam file="$(find wpb_home_tutorials)/nav_lidar/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find wpb_home_tutorials)/nav_lidar/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find wpb_home_tutorials)/nav_lidar/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find wpb_home_tutorials)/nav_lidar/global_costmap_params.yaml" command="load" />
    <param name="base_global_planner" value="global_planner/GlobalPlanner" /> 
    <param name="use_dijkstra" value="true"/>
    <param name="base_local_planner" value="wpbh_local_planner/WpbhLocalPlanner" />

    <param name= "controller_frequency" value="10" type="double"/>
  </node>
  <!-- Run TTS node -->
  <node pkg="xfyun_waterplus" type="tts_node" name="xfyun_tts_node" /> 
  <!-- Run sound_play -->
  <node name="soundplay_node" pkg="sound_play" type="soundplay_node.py"/>

  <!-- Run script -->
  <node pkg="fri01_apps" type="fri01_basic" name="fri01_basic" output="screen"/>

</launch>
